### PR TITLE
Updates docker file to create test server package

### PR DESCRIPTION
**Pull request by bhess**
The update uses..
- liboqs 0.9.0
- latest oqs-provider (main)
- latest nginx 1.25.2

Also updates the nginx docker demo to v1.25.2.
Note that with [nginx 1.25.2](http://nginx.org/en/CHANGES), the location of the openssl config file needs to be passed as an env variable. This is reflected in the Dockerfile of the nginx demo.

The updated package is already deployed at test.openquantumsafe.org.
A test run with `testrun.sh` passes.
I'll update the server again to a tagged oqs-provider release once available.

### CODE REVIEW

File: nginx/fulltest-provider/README.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ Note that, the oqs-nginx-{LIBOQS_VERSION}.tgz package contains all required conf
 
 #### Activation
 
-Execute `/opt/nginx/sbin/nginx -c /opt/nginx/interop.conf` to start the test server.
+Execute `OPENSSL_CONF=/opt/openssl/.openssl/ssl/openssl.cnf /opt/nginx/sbin/nginx -c /opt/nginx/interop.conf` to start the test server.
```
Comment (baentsch) [inline nginx/fulltest-provider/README.md:35]: This doesn't look like an "under-the-hood" change invisible to users: Doesn't this warrant updating at least USAGE.md for `nginx`? Why is this apparently user-API changing version update necessary?

File: nginx/Dockerfile [hunk 2]
```diff
@@ -126,6 +126,5 @@ STOPSIGNAL SIGTERM
 # Enable a normal user to create new server keys off set CA
 RUN addgroup -g 1000 -S oqs && adduser --uid 1000 -S oqs -G oqs && chown -R oqs.oqs ${INSTALLDIR}
 USER oqs
-
+ENV OPENSSL_CONF ${OSSLDIR}/ssl/openssl.cnf
```
Comment (baentsch) [inline nginx/Dockerfile:129]: This bypasses all `nginx` config logic. Why is this necessary? Shouldn't this be documented somewhere?

File: nginx/Dockerfile [hunk 3]
```diff
@@ -18,10 +18,10 @@ ARG SIG_ALG="dilithium3"
 ARG DEFAULT_GROUPS=x25519:x448:kyber512:p256_kyber512:kyber768:p384_kyber768:kyber1024:p521_kyber1024
 
 # define the nginx version to include
-ARG NGINX_VERSION=1.23.3
+ARG NGINX_VERSION=1.25.2
 
 # Define the degree of parallelism when building the image; leave the number away only if you know what you are doing
-ARG MAKE_DEFINES="-j 18"
+ARG MAKE_DEFINES="-j"
```
Comment (baentsch) [inline nginx/Dockerfile:24]: This change will make any "few cores" machine running this squeak, possibly crash (have seen this before). I appreciate that "modern laptops" couldn't care less but we don't prescribe this for building the demos. Also, some CI systems (notably CCI) has problems with unlimited "-j". The 18 was what CCI could handle without failures. Unless you can point to a document stating improved CCI capabilities, please revert.

**Review by baentsch — CHANGES_REQUESTED** [review]
See single comments

**Comment by baentsch** [comment]
> I'll update the server again to a tagged oqs-provider release once available.

As a heads-up: This may take some more time than anticipated due to an [upstream error crashing oqs-provider CI](https://github.com/open-quantum-safe/oqs-provider/actions/runs/6568599921/job/17846906799?pr=288)...

File: nginx/Dockerfile (hunk 3 above)
Comment (bhess) [reply nginx/Dockerfile]: Sorry I didn't intend to make the machines and CI squeak :) Reverted in 25ea3f839e987388c2346b24e806a8a4adcf5bd2

**Comment by bhess** [comment]
Thanks for the review @baentsch and the heads-up! Please see the update and the single comments.

**Review by baentsch — APPROVED** [review]
Thanks for the updates, @bhess. LGTM.

**Comment by baentsch** [comment]
> I'll update the server again to a tagged oqs-provider release once available.

@bhess FWIW [oqsprovider release is available](https://github.com/open-quantum-safe/oqs-provider/releases/tag/0.5.2)...

**Comment by bhess** [comment]
> > I'll update the server again to a tagged oqs-provider release once available.
> 
> @bhess FWIW [oqsprovider release is available](https://github.com/open-quantum-safe/oqs-provider/releases/tag/0.5.2)...

Thanks @baentsch! Will update the server today.

**Comment by pi-314159** [comment]
@bhess Should we also update tags in README.md?

> Open Quantum Safe TLS/X.509 interoperability test server at https://test.openquantumsafe.org/ (set up using oqs-provider **v0.5.0** and liboqs **v0.8.0**)

**Comment by bhess** [comment]
> @bhess Should we also update tags in README.md?

Added an update in PR #241. Thanks @pi-314159!
